src/pages/dashboard/index.page.tsx

```diff
@@ -166,54 +184,264 @@ const StatusIndicator = ({
+  const [images, setImages] = useState<ImageType[]>([]);
+  const [imageToEditId, setImageToEditId] = useState('');
+  const [isImageUploading, setIsImageUploading] = useState(false);
+  const [isImageHovered, setIsImageHovered] = useState(false);
+  const weights = getScopeWeights(scope);
+  const minimumScore = useMemo(() => getMinimumScore(weights), [weights]);
+  const imageToEdit = useMemo(() => {
+    const image = images.find((image) => image.parsedId === imageToEditId);
+
+    if (!image) return null;
+
+    const { file, ...imageWithoutFile } = image;
+
+    return imageWithoutFile;
+  }, [images, imageToEditId]);
+
+  const addImageToEventMutation = useAddOfferImageMutation({
+    onSuccess: () => {
+      setIsPictureUploadModalVisible(false);
+      setTimeout(async () => {
+        await queryClient.invalidateQueries('events');
+        setIsImageUploading(false);
+      }, 1000);
+    },
+  });
+
+  const handleSuccessAddImage = ({ imageId }) => {
+    return addImageToEventMutation.mutate({ eventId, imageId, scope });
+  };
+
+  const addImageMutation = useAddImageMutation({
+    onSuccess: handleSuccessAddImage,
+  });
+
+  const updateOfferImageMutation = useUpdateOfferImageMutation({
+    onSuccess: async () => {
+      setIsPictureUploadModalVisible(false);
+    },
+  });
+
+  const handleSubmitValid = async ({
+    file,
+    description,
+    copyrightHolder,
+  }: FormData) => {
+    try {
+      setIsImageUploading(true);
+      if (imageToEdit) {
+        await updateOfferImageMutation.mutateAsync({
+          eventId,
+          imageId: imageToEdit.parsedId,
+          description,
+          copyrightHolder,
+          scope,
+        });
+
+        return;
+      }
+
+      await addImageMutation.mutateAsync({
+        description,
+        copyrightHolder,
+        file: file?.[0],
+        language: i18n.language,
+      });
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
   return (
-    <Inline
-      flex={1}
-      css={css`
-        display: grid;
-        gap: ${parseSpacing(4)};
-        grid-template-columns: ${status ? '5fr 3fr 1fr' : '8fr 2fr'};
-      `}
-      {...getInlineProps(props)}
-    >
-      <Stack spacing={2}>
-        <Inline spacing={3}>
-          <Link href={url} color={getValue('listItem.color')} fontWeight="bold">
-            {title}
-          </Link>
-        </Inline>
-        <Text>{description}</Text>
-      </Stack>
-      {status && <StatusIndicator {...status} />}
-      <Inline justifyContent="flex-end" minWidth="11rem">
-        {finishedAt ? (
-          <Text color={getValue('listItem.passedEvent.color')}>
-            {finishedAt}
-          </Text>
-        ) : (
-          actions.length > 0 && (
-            <Dropdown variant={DropDownVariants.SECONDARY} isSplit>
-              {actions}
-            </Dropdown>
-          )
+    <Inline spacing={5} {...getInlineProps(props)} flex={1}>
+      <Inline width="100" alignItems="center">
```

Comment:
There's a lot of code on this page.
Maybe we could abstract the the offer row of the dashboard into it's own component?